benchmark/BenchmarkMain.cpp

```diff
@@ -218,4 +206,99 @@ int main() {
       addTablesToStringstream);
 
   std::cout << visualization.str() << "\n";
-};
+}
+
+/*
+ * @brief Create a nlohmann::json object with all relevant informations
+ *  about the measurments taken by a BenchmarkRecords.
+ */
+nlohmann::json benchmarksToJson(const BenchmarkRecords& records){
+  // The values for all the categories of benchmarks.
+  const std::vector<BenchmarkRecords::RecordEntry>& singleMeasurements =
+    records.getSingleMeasurements();
+  const std::vector<BenchmarkRecords::RecordGroup>& recordGroups =
+    records.getGroups();
+  const std::vector<BenchmarkRecords::RecordTable>& recordTables =
+    records.getTables();
+
+  // Creating the json object. We actually don't want BenchmarkRecords to
+  // be serilized, because that is the management class for measured
+  // benchmarks. We just want the measured benchmarks.
+  return nlohmann::json{{"singleMeasurements", singleMeasurements},
+    {"recordGroups", recordGroups}, {"recordTables", recordTables}};
+}
+
+void writeJsonToFile(nlohmann::json j, std::string fileName) {
+  // Using this constructor, the file is already opened.
+  std::ofstream myFile(fileName, std::ios::out);
+
+  // Error message, if the file couldn't be opened.
+  if (!myFile.is_open()){
+    std::cerr << "Unable to open file '" << fileName << "'.";
+  }
+
+  // nlohmann already has support for things like this.
+  myFile << j;
+
+  myFile.close();
+}
+
+}
+
+/*
+ * @brief Goes through all types of registered benchmarks, measures their time
+ * and prints their measured time in a fitting format.
+ */
+int main(int argc, char** argv) {
+  // Prints how to use the file correctly and exits.
+  auto printUsageAndExit = [&argv](){
+    std::cerr << "Usage: ./" << argv[0] << " [options]\n"
+      << " --help, -h: Print this help message.\n"
+      << " --print, -p: Roughly prints all benchmarks.\n"
+      << " --json, -j <file>: Writes the benchmarks as json to a file.\n";
+    exit(1);
```

Comment:
Such a manual Usage string easily gets out of sync.
You can print the `program_options` directly, that way you always have the correct usage string. Have a look at `IndexBuilderMain` or `ServerMain` for how this can be done.

Reply:
Done, see [b85ec0a](https://github.com/ad-freiburg/qlever/pull/860/commits/b85ec0ab2c3b5e23eb68293e7bdd5218a2b5a2e8).